micrometer-observation/src/main/java/io/micrometer/observation/docs/ObservationDocumentation.java

```diff
@@ -143,7 +141,7 @@ default Observation observation(ObservationRegistry registry) {
      * @param context observation context
      * @return observation
      */
-    default Observation observation(ObservationRegistry registry, @Nullable Observation.Context context) {
+    default Observation observation(ObservationRegistry registry, @Nullable Supplier<Observation.Context> context) {
```

Comment:
FIXME: The Supplier shouldn't be `@Nullable` in my opinion.

Reply:
+1, but probably we can provide a supplier that return null, i.e.: `Context.NULL`, see my previous comment.